test/ASTGen/autodiff.swift

```diff
@@ -0,0 +1,109 @@
+// RUN: %empty-directory(%t)
+
+// RUNx: %target-swift-frontend-dump-parse \
+// RUNx:   -enable-experimental-feature ParserASTGen \
+// RUNx:   | %sanitize-address > %t/astgen.ast
+
+// RUNx: %target-swift-frontend-dump-parse \
+// RUNx:   | %sanitize-address > %t/cpp-parser.ast
+
+// RUNx: %diff -u %t/astgen.ast %t/cpp-parser.ast
+
+// RUN: %target-typecheck-verify-swift
+
+import _Differentiation
```

Comment:
This test might want to add `REQUIRED: differentiable_programming` or be moved into the `test/AutoDiff/` directory.